lib/ui/semantics.dart

```diff
@@ -641,6 +641,176 @@ class SemanticsFlag {
   }
 }
 
+/// A string that carries a list of [StringAttribute]s.
+class AttributedString extends NativeFieldWrapperClass2 {
+  /// Creates a attributed string.
+  ///
+  /// The [TextRange] in the [attributes] must be inside the length of the
+  /// [string].
+  ///
+  /// The [attributes] must not be changed after the attributed string is
+  /// created.
+  AttributedString(
+    this.string, {
+    this.attributes = const <StringAttribute>[],
+  }) : assert(() {
+         for (final StringAttribute attribute in attributes) {
+           assert(
+             string.length >= attribute.range.start &&
```

Comment:
Do they both have to be >= 0 as well?

Reply:
How could string.length be > 0?

Reply:
I was talking about attribute.range.start/end >= 0.